src/utils/utils.ts

```diff
@@ -31,3 +32,15 @@ export async function executeCommand(command: string, cwd?: string): Promise<str
 export function generateQuotedFullName(schema: string, objectName: string): string {
 	return `[${escapeClosingBrackets(schema)}].[${escapeClosingBrackets(objectName)}]`;
 }
+
+/**
+ * Promise with timeout to check for another promise fulfillment
+ * @returns a promise that resolves after the specified timeout
+ */
+export function waitForPromise(): Promise<void> {
+	return new Promise((_, reject) => {
+		setTimeout(() => {
+			reject(new Error(constants.timeoutSqlBindingError));
```

Comment:
This message is pretty specific to sql bindings. I'd suggest just having this take in the message as a parameter